src/org/zaproxy/zap/extension/frontendscanner/files/scripts/scripts/client-side-passive/scan-jwt-tokens.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * When storage is written to, check if the written value is a JWT token.
+ */
+
+frontEndScanner.mailbox.subscribe('storage', (_, storageEntry) => {
+  // A storage entry has the following form:
+  // {key: 'foo', value: 'bar', action: 'get|remove|set'}
+  if (storageEntry.action === 'set') {
+    verify(storageEntry);
+    verifySubEntries(storageEntry);
+  }
+});
+
+function verify(storageEntry) {
+  const potentialToken = storageEntry.value;
+  const tokenContent = _parseJwt(potentialToken);
+
+  if (!tokenContent) { return; }
+
+  _reportJwtToken(storageEntry, tokenContent);
+
+  if (tokenContent.header.alg === 'HS256') {
+    _reportWeakAlgorithm(storageEntry, tokenContent);
+  } else if (tokenContent.header.alg === 'none') {
```

Comment:
or not present?

Reply:
`tokenContent` is the result of `_parseJwt(token)` which returns an object containing a `header` key.
If the decoding fail in `_parseJwt(token)`, `false` is returned by it, this function is left earlier (L18) and this line is never reached.